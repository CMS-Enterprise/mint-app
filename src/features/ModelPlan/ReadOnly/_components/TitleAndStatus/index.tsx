import React from 'react';
import { useTranslation } from 'react-i18next';
import { Icon } from '@trussworks/react-uswds';
import classNames from 'classnames';
import { TaskListStatusTag } from 'features/ModelPlan/TaskList/_components/TaskListItem';
import {
  DataExchangeApproachStatus,
  ModelStatus,
  MtoStatus,
  TaskStatus
} from 'gql/generated/graphql';

import UswdsReactLink from 'components/LinkWrapper';
import { formatDateLocal } from 'utils/date';

type TitleAndStatusProps = {
  modelID: string;
  clearance: boolean | undefined;
  clearanceTitle: string;
  heading: string;
  subHeading?: string;
  isViewingFilteredView?: boolean;
  status: ModelStatus | TaskStatus | DataExchangeApproachStatus | MtoStatus;
  modifiedOrCreatedDts?: string | null;
  editDates?: boolean;
};

const TitleAndStatus = ({
  modelID,
  clearance,
  clearanceTitle,
  heading,
  subHeading,
  isViewingFilteredView,
  status,
  modifiedOrCreatedDts,
  editDates
}: TitleAndStatusProps) => {
  const { t } = useTranslation('modelPlanTaskList');
  const { t: h } = useTranslation('generalReadOnly');
  const { t: timelineMiscT } = useTranslation('timelineMisc');

  return (
    <div>
      <div className="display-flex margin-bottom-2 ">
        <h2 className="margin-y-0 margin-right-2">
          {clearance ? clearanceTitle : heading}
        </h2>

        {subHeading && (
          <p className="mint-body-large text-base-dark margin-y-0 flex-align-self-end">
            {subHeading}
          </p>
        )}
      </div>

      {!isViewingFilteredView && status && (
        <div className="display-flex flex-align-center flex-wrap margin-right-1 margin-bottom-4">
          <p className="margin-y-0 text-bold margin-right-1">{t('status')}</p>

          <TaskListStatusTag
            status={status}
            classname="width-fit-content margin-right-1"
          />

          <div className="display-flex flex-align-center flex-wrap margin-right-1">
            {!!modifiedOrCreatedDts && (
              <p
                className={classNames(
                  'margin-y-0 text-normal margin-right-1 text-base-dark'
                )}
              >
                {h('lastUpdate')}
                {formatDateLocal(modifiedOrCreatedDts, 'MM/dd/yyyy')}
              </p>
            )}

            {editDates && (
              <>
                <div className="border-left-2px border-base-light margin-left-105 padding-left-2 height-2 margin-top-1px" />

                <Icon.Edit
                  className="margin-right-1 text-primary"
                  data-testid="edit-icon"
                />

                <UswdsReactLink
                  data-testid="edit-dates-link"
                  to={`/models/${modelID}/collaboration-area/model-timeline`}
                >
                  {timelineMiscT('editDates')}
                </UswdsReactLink>
              </>
            )}
          </div>
        </div>
      )}
    </div>
  );
};

export default TitleAndStatus;
