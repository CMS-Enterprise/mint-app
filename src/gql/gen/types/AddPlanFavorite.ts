/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL mutation operation: AddPlanFavorite
// ====================================================

export interface AddPlanFavorite_addPlanFavorite {
  __typename: "PlanFavorite";
  modelPlanID: UUID;
  userID: UUID;
}

export interface AddPlanFavorite {
  addPlanFavorite: AddPlanFavorite_addPlanFavorite;
}

export interface AddPlanFavoriteVariables {
  modelPlanID: UUID;
}
