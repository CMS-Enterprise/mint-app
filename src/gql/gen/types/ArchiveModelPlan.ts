/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL mutation operation: ArchiveModelPlan
// ====================================================

export interface ArchiveModelPlan_updateModelPlan {
  __typename: "ModelPlan";
  id: UUID;
}

export interface ArchiveModelPlan {
  updateModelPlan: ArchiveModelPlan_updateModelPlan;
}

export interface ArchiveModelPlanVariables {
  id: UUID;
  archived: boolean;
}
