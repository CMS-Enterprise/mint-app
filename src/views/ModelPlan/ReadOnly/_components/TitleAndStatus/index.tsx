import React from 'react';
import { TaskStatus } from 'gql/gen/graphql';

import { TaskListStatusTag } from 'views/ModelPlan/TaskList/_components/TaskListItem';

type TitleAndStatusProps = {
  clearance: boolean | undefined;
  clearanceTitle: string;
  heading: string;
  isViewingFilteredView?: boolean;
  status: TaskStatus | undefined;
};

const TitleAndStatus = ({
  clearance,
  clearanceTitle,
  heading,
  isViewingFilteredView,
  status
}: TitleAndStatusProps) => {
  return (
    <div className="display-flex flex-justify flex-align-start">
      <h2 className="margin-top-0 margin-bottom-4">
        {clearance ? clearanceTitle : heading}
      </h2>
      <div>
        {!isViewingFilteredView && status && (
          <TaskListStatusTag status={status} />
        )}
      </div>
    </div>
  );
};

export default TitleAndStatus;
