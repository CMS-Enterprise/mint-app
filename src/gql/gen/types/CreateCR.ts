/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { PlanCRCreateInput } from "./../../../types/graphql-global-types";

// ====================================================
// GraphQL mutation operation: CreateCR
// ====================================================

export interface CreateCR_createPlanCR {
  __typename: "PlanCR";
  id: UUID;
  modelPlanID: UUID;
  idNumber: string;
  dateInitiated: Time;
  dateImplemented: Time | null;
  title: string;
  note: string | null;
}

export interface CreateCR {
  createPlanCR: CreateCR_createPlanCR;
}

export interface CreateCRVariables {
  input: PlanCRCreateInput;
}
