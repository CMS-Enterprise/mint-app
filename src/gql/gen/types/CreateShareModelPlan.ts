/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { ModelViewFilter } from "./../../../types/graphql-global-types";

// ====================================================
// GraphQL mutation operation: CreateShareModelPlan
// ====================================================

export interface CreateShareModelPlan {
  shareModelPlan: boolean;
}

export interface CreateShareModelPlanVariables {
  modelPlanID: UUID;
  viewFilter?: ModelViewFilter | null;
  usernames: string[];
  optionalMessage?: string | null;
}
