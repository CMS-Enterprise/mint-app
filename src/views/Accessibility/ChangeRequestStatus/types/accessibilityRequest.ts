/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { AccessibilityRequestStatus } from "./../../../../types/graphql-global-types";

// ====================================================
// GraphQL query operation: accessibilityRequest
// ====================================================

export interface accessibilityRequest_accessibilityRequest_statusRecord {
  __typename: "AccessibilityRequestStatusRecord";
  status: AccessibilityRequestStatus;
}

export interface accessibilityRequest_accessibilityRequest {
  __typename: "AccessibilityRequest";
  id: UUID;
  name: string;
  statusRecord: accessibilityRequest_accessibilityRequest_statusRecord;
}

export interface accessibilityRequest {
  accessibilityRequest: accessibilityRequest_accessibilityRequest | null;
}

export interface accessibilityRequestVariables {
  id: UUID;
}
