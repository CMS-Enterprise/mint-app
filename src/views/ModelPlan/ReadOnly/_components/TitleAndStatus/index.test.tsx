import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';

import { TaskStatus } from 'types/graphql-global-types';

import TitleAndStatus from './index';

describe('Title and Status component for Read Only Pages', () => {
  it('renders without crashing', async () => {
    render(
      <TitleAndStatus
        clearance={false}
        clearanceTitle="Clearance"
        heading="Regular Heading"
        isViewingFilteredView={false}
        status={TaskStatus.IN_PROGRESS}
      />
    );

    await waitFor(() => {
      expect(screen.getByText('Regular Heading')).toBeInTheDocument();
      expect(screen.getByText('In progress')).toBeInTheDocument();
    });
  });

  it('renders Clearance heading', async () => {
    render(
      <TitleAndStatus
        clearance
        clearanceTitle="Clearance"
        heading="Regular Heading"
        isViewingFilteredView={false}
        status={TaskStatus.IN_PROGRESS}
      />
    );

    await waitFor(() => {
      expect(screen.getByText('Clearance')).toBeInTheDocument();
      expect(screen.getByText('In progress')).toBeInTheDocument();
    });
  });

  it('renders the correct status', async () => {
    render(
      <TitleAndStatus
        clearance
        clearanceTitle="Clearance"
        heading="Regular Heading"
        isViewingFilteredView={false}
        status={TaskStatus.READY_FOR_CLEARANCE}
      />
    );

    await waitFor(() => {
      expect(screen.getByText('Clearance')).toBeInTheDocument();
      expect(screen.getByText('Ready for clearance')).toBeInTheDocument();
    });
  });

  it('does not render status for Filtered View groups', () => {
    render(
      <TitleAndStatus
        clearance
        clearanceTitle="Clearance"
        heading="Regular Heading"
        isViewingFilteredView
        status={TaskStatus.READY_FOR_CLEARANCE}
      />
    );

    expect(screen.getByText('Clearance')).toBeInTheDocument();
    expect(screen.queryByTestId('tasklist-tag')).not.toBeInTheDocument();
  });

  it('matches snapshot', async () => {
    const { asFragment } = render(
      <TitleAndStatus
        clearance={false}
        clearanceTitle="Clearance"
        heading="Regular Heading"
        isViewingFilteredView={false}
        status={TaskStatus.IN_PROGRESS}
      />
    );

    await waitFor(() => {
      expect(screen.getByText('Regular Heading')).toBeInTheDocument();
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
