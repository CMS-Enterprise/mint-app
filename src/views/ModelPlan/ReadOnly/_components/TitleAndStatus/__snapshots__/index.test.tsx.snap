// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title and Status component for Read Only Pages matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="display-flex flex-justify flex-align-start"
  >
    <h2
      class="margin-top-0 margin-bottom-4"
    >
      Regular Heading
    </h2>
    <div>
      <div
        class="model-plan-task-list__task-tag line-height-body-1 text-bold mint-no-print bg-warning"
        data-testid="tasklist-tag"
      >
        <span>
          In progress
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
